phoenix-core/src/main/java/org/apache/phoenix/mapreduce/OrphanViewTool.java

```diff
@@ -0,0 +1,812 @@
+        }
+        if (cmdLine.hasOption(HELP_OPTION.getOpt())) {
+            printHelpAndExit(options, 0);
+        }
+        if (cmdLine.hasOption(OUTPUT_PATH_OPTION.getOpt()) && cmdLine.hasOption(INPUT_PATH_OPTION.getOpt())) {
+            throw new IllegalStateException("Specify either " + OUTPUT_PATH_OPTION.getLongOpt() + " or "
+                    + INPUT_PATH_OPTION.getOpt());
+        }
+        if (cmdLine.hasOption(INPUT_PATH_OPTION.getOpt()) && !cmdLine.hasOption(CLEAN_ORPHAN_VIEWS_OPTION.getOpt())) {
+            throw new IllegalStateException(INPUT_PATH_OPTION.getLongOpt() + " is only used with "
+                    + IDENTIFY_ORPHAN_VIEWS_OPTION.getOpt());
+        }
+        if (cmdLine.hasOption(IDENTIFY_ORPHAN_VIEWS_OPTION.getOpt()) && cmdLine.hasOption(CLEAN_ORPHAN_VIEWS_OPTION.getOpt())) {
+            throw new IllegalStateException("Specify either " + IDENTIFY_ORPHAN_VIEWS_OPTION.getLongOpt() + " or "
+                    + IDENTIFY_ORPHAN_VIEWS_OPTION.getOpt());
+        }
+        if (cmdLine.hasOption(OUTPUT_PATH_OPTION.getOpt()) && (!cmdLine.hasOption(IDENTIFY_ORPHAN_VIEWS_OPTION.getOpt()) &&
+                !cmdLine.hasOption(CLEAN_ORPHAN_VIEWS_OPTION.getOpt()))) {
+            throw new IllegalStateException(OUTPUT_PATH_OPTION.getLongOpt() + " requires either " +
+                    IDENTIFY_ORPHAN_VIEWS_OPTION.getOpt() + " or " + CLEAN_ORPHAN_VIEWS_OPTION.getOpt());
+        }
+        if (cmdLine.hasOption(CLEAN_ORPHAN_VIEWS_OPTION.getOpt())) {
+            clean = true;
+        }
+        else if (!cmdLine.hasOption(IDENTIFY_ORPHAN_VIEWS_OPTION.getOpt())) {
+            throw new IllegalStateException("Specify either " +
+                    IDENTIFY_ORPHAN_VIEWS_OPTION.getOpt() + " or " + CLEAN_ORPHAN_VIEWS_OPTION.getOpt());
+        }
+        if (cmdLine.hasOption(AGE_OPTION.getOpt())) {
+            age = Long.valueOf(cmdLine.getOptionValue(AGE_OPTION.getOpt()));
+        }
+
+        outputPath = cmdLine.getOptionValue(OUTPUT_PATH_OPTION.getOpt());
+        inputPath = cmdLine.getOptionValue(INPUT_PATH_OPTION.getOpt());
+    }
+
+    private void printHelpAndExit(String errorMessage, Options options) {
+        System.err.println(errorMessage);
+        printHelpAndExit(options, 1);
+    }
+
+    private void printHelpAndExit(Options options, int exitCode) {
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.printHelp("help", options);
+        System.exit(exitCode);
+    }
+
+    private class Key {
+        private String tenantId;
+        private String schemaName;
+        private String tableName;
+
+        public Key (String tenantId, String schemaName, String tableName) {
+            this.tenantId = tenantId;
+            this.schemaName = schemaName;
+            this.tableName = tableName;
+        }
+
+        public Key (String tenantId, String fullTableName) {
+            this.tenantId = tenantId;
+            String[] columns = fullTableName.split("\\.");
```

Comment:
You should use existing Phoenix APIs which parse the fullTableName using either '.' or ':' based on the value of the `isNamespaceMapped` property. Check out `SchemaUtil.java`

Reply:
Good catch. I will do that.

Reply:
@kadirozde I was referring to Phoenix schema to HBase namespace mapping, but you are not trying to ever access the physical HBase table. I missed that. As per our offline discussion, this should be fine.